### ISSUE TITLE
Chap10.pynb のコード修正ポイント

### ISSUE THREAD

**Issue by build1024** [koshian2/MosaicDeeplearningBook#14]
お世話になっております。
10章の演習問題をTensorFlow 2.6.0で動作させる際に必要だった修正箇所をご報告いたします。
各項目はすでに他のIssueで挙げられていますが、後から探しやすいように別のIssueとして整理しておきます。

# import
#12, #13 と同様
```python
#from utils_tf20 import Reduction, distributed, make_grid, make_citation
#import utils_tf20 as utils
from utils import Reduction, distributed, make_grid, make_citation
import utils
```

# Hinge Loss
#10 と同様
```python
            #minval = tf.minimum(logits - 1, tf.zeros(logits.shape, dtype=logits.dtype))
            minval = tf.minimum(logits - 1, tf.zeros(tf.shape(logits), dtype=logits.dtype))
            minval = tf.minimum(logits - 1, 0.) # 実はこれでもOK
```
```python
            #minval = tf.minimum(-logits - 1, tf.zeros(logits.shape, dtype=logits.dtype))
            minval = tf.minimum(-logits - 1, tf.zeros(tf.shape(logits), dtype=logits.dtype))
            minval = tf.minimum(-logits - 1, 0.) # 実はこれでもOK
```

# データセットの作成
#4 ほかと同様
```python
    trainset = tf.data.Dataset.from_tensor_slices(trains).map(
        lambda *args: [preprocess(a) for a in args]
    #).shuffle(128).batch(batch_size).prefetch(buffer_size=tf.data.experimental.AUTOTUNE)    
    ).shuffle(128).batch(batch_size, drop_remainder=True).prefetch(buffer_size=tf.data.experimental.AUTOTUNE)    
```
```python
    valset = valset.map(
        lambda *args: [preprocess(a) for a in args]
    #).batch(batch_size).prefetch(buffer_size=tf.data.experimental.AUTOTUNE)    
    ).batch(batch_size, drop_remainder=True).prefetch(buffer_size=tf.data.experimental.AUTOTUNE)    
```

以上3箇所の変更で5エポックの学習が完了しました。
よろしくお願いいたします。
